Compass129Sources/COMfile/C12COM9.ASC

```diff
@@ -973,37 +973,37 @@ m_vgl_HLDE	LD	A,H
 	SUB	E
 	RET
 
-m_freeall	ld	a,#b6	;poke nieuwe opcode: res 6,(ix+0)
-	ld	(m_res6),a	;hierdoor worden ook de gebruikte
+m_freeall	ld	a,#b6	;poke new opcode: res 6,(ix+0)
+	ld	(m_res6),a	;as a result, the used
 m_free	LD	A,(stat_mem)
 	OR	A
 	scf
-	RET	Z	;enkel als er een memmanager is
+	RET	Z	;only if there is a mem manager
 	LD	IX,buffer-2
 m_freenext	INC	IX
 	INC	IX
-	db	#DD,#CB,0	;res 5,(ix+0):dit doet dus niets
-m_res6	db	#AE	;om ALLES te free-en: poke hier #B6
-	LD	A,(IX+#00)	;dit is opcode voor res 6,(ix+0)
+	db	#DD,#CB,0	;res 5,(ix+0): so this does nothing
+m_res6	db	#AE	;to free-en EVERYTHING: poke here #B6
+	LD	A,(IX+#00)	;this is opcode for res 6,(ix+0)
 	OR	A
 	scf
-	RET	Z	;gedaan
-	bit	4,a	;is het een tpasegment?
-	jr	nz,m_freenext	;ja, niet vrijgeven dan
-	BIT	6,A	;segmenten in use niet vrijgeven en ze
-	JR	NZ,m_bit6seg	;onder memman de reserved status geven
-	ld	b,a	;slotcode in b
-	LD	c,(IX+#01)	;pagenr in c
+	RET	Z	;done
+	bit	4,a	;is it a tpa segment?
+	jr	nz,m_freenext	;yes, don't free it then
+	BIT	6,A	;do not free segments in use and they
+	JR	NZ,m_bit6seg	;give the reserved status under memman

```

Comment:
"do not free segments that are in use, and give them the reserved status within/under memman"